templates/prescription/general_list.html

```diff
@@ -83,6 +83,34 @@
     }
 }
 
+function deleteDrug(d) {
+    let msg = 'Do you really want to delete ';
+    let choice = confirm(msg + d);
+    let isPromise = top.jsFetchGlobals('e_rx');
+    isPromise.then(
+        msg => {alert('msg.e_rx.deleteWarn');
+            if (choice == true) {
+                $.ajax({
```

Comment:
You need to add the translation message in library/ajax/phpvars_to_js.php
something like $custom['e_rx']['deleteWarn'] = xlt("Do you really want to delete");
This is one that can be reused.

also alert() should be on it's own line.

You should be able to use the js xl(). Just include library/js/utility.js in your form if not already in scope.
let msg = xl('Do you really want to delete '); or you may want for 5.0.2 so as is is okay...

Reply:
I am sticking with 5.0.3. The 5.0.2 stuff works for our production. Just pushing this up so that others can enjoy the benefits. 

Now, it seems like you are giving me two options. I am taking the first and edited the phpvars_to_js.php. As, I agree with the path that is on and it is reusable. In the csrf, If it fails, an error will be thrown. This should pass back to the requester.  

So, I changed the order of operations and filtered the $_POST['drugId'] to make sure it is an int first then took your suggestion to trim the int then make sure the $id is set before going on to the SQL portion.  The SQL is wrapped in a try-catch block translating Done when complete.

Reply:
Guessing that phpvars_to_js.php use to support translations will disappear at some point now that we have the javascript xl function, so would rec not going that route. Options would be:
1. If want to work only in 5.0.3, then use xl
```
let msg = xl('Do you really want to delete');
```
2. If want to work in both both 5.0.2/5.0.3, then use the smarty translations/escaping mechanism:
```
let msg = {/literal}{xlj t='Do you really want to delete'{literal}};
```

Reply:
> Guessing that phpvars_to_js.php use to support translations will disappear at some point now that we have the javascript xl function

I thought about this yesterday and almost removed. The main reason I added this function was for accessing globals and session variables used as flags from pure JS etc.
I don't know how many times i've been writing JS and wish I knew if certain php vars are present.
Though we could move out of top to utility.js.